test/issues/github_412.cpp

```diff
@@ -0,0 +1,17 @@
+#include "boost/hana.hpp"
```

Comment:
We use `<>` to include Hana's headers in the tests.  Also, we try to include the most specific header. `#include <boost/hana/functional/overload.hpp>`